pkg/operator/ceph/file/filesystem.go

```diff
@@ -232,3 +255,17 @@ func downFilesystem(context *clusterd.Context, namespace, filesystemName string)
 	logger.Infof("Downed filesystem %s", filesystemName)
 	return nil
 }
+
+// generateMetadataPoolName is a helper function that generates a new name for the given metadata pool and returns the renamed pool.
+func generateMetadataPoolName(fsName string, metadataPool *model.Pool) *model.Pool {
```

Comment:
Model pool is gone, we must rebase and adapt your code.

Reply:
Yes, and that is why I believe that I'm not able to use `*&pool.Name` anymore.
But what kind of changes needs to be adapted?.

Reply:
You just need to use values from the spec.

Reply:
Ok, I looked into this and I think from the spec you meant `cephv1.FilesystemSpec`.
I made changes according to that.
Let me know if I missed anything.
thanks